Code to solve the problems described in Readme and tests associated with solution.

GauthamPrakash14 opened the pull request:
**Code to solve the problems given in the Readme and also the test cases for the solution.**

Changes in repo. 
Contains solutions to the following problems given in the Readme. 
_**1. Find journeys that are 90 minutes or more.
2. Find the average speed per journey in kph.
3. Find the total mileage by driver for the whole day.
4. Find the most active driver - the driver who has driven the most 
5. Output results to STDOUT.**_

The new code also tackles the following. 
**_1. Make the code easier to understand and more maintainable. 
2. Come up with good tests that prove the code generates correct results.
3. Come up with good abstractions on the input data that make the solution easy to understand._**

Reviewers:
Jorge Ramos @jAlpedrinha

jAlpedrinha reviewed (changes requested):
Thank you for your PR

Left a few comments on the code.

I noticed you used sbt and bloop, but the requirements of this challenge require you to use mvn. Could you make the command `mvn clean install` work ?

GauthamPrakash14 commented:
> Thank you for your PR
> 
> Left a few comments on the code.
> 
> I noticed you used sbt and bloop, but the requirements of this challenge require you to use mvn. Could you make the command `mvn clean install` work ?

GauthamPrakash14 commented on src/main/java/ai/humn/telematics/ProcessDataFile.scala:
```diff
@@ -2,57 +2,156 @@ package ai.humn.telematics
+
+  def main(args: Array[String]): Unit = {
+
+    val filePath = "/Users/gautham/Interview assessment/AON/de-coding-test-gautham-prakash/src/test/resources/2021-10-05_journeys.csv" // Replace this file path with the CSV file path in your repo.
```
Passing the file through main args is somehow not working in my workspace. That is the reason why I choose to read the file directly from a file path. 

I was blocked, so I defined a separate function to read the file. I have now added and commented out the code to read the file from args. Please could you check from your end also by de-commenting and then commenting the function call and line below?

GauthamPrakash14 commented on src/main/java/ai/humn/telematics/ProcessDataFile.scala:
```diff
@@ -2,57 +2,156 @@ package ai.humn.telematics
+  def main(args: Array[String]): Unit = {
+
+    val filePath = "/Users/gautham/Interview assessment/AON/de-coding-test-gautham-prakash/src/test/resources/2021-10-05_journeys.csv" // Replace this file path with the CSV file path in your repo.
+    val j = getData(filePath)
```
Resolved this in the latest commit.

GauthamPrakash14 commented on src/main/java/ai/humn/telematics/ProcessDataFile.scala:
```diff
@@ -2,57 +2,156 @@ package ai.humn.telematics
 
-    // Make a variable to hold the parsed lines from the file.
-    var results = ListBuffer[ Array[String] ]()
+    var results = ListBuffer[ Array[String]]()
```
I could think of using a custom class with type-defined variables to store the journeys.

Following is an example implementation of custom class. 
class Journey(val startLocation: String, val endLocation: String, val startTime: String, val endTime: String)

Advantages of using custom class is we get to control the type of variable and encounter bugs.

GauthamPrakash14 commented:
#Update. 

1. Resolved name of the variable to a more meaningful name. 
2. removed bloop and sbt dependencies from the project and made sure `mvn clean install` is working. 
3. added code to read files from agrs and commented code since reading from args is not working in my workspace. (Please refer to the above conversation).
